src/OpenLoco/Network/Socket.cpp

```diff
@@ -0,0 +1,628 @@
+#include <atomic>
+#include <cassert>
+#include <chrono>
+#include <cmath>
+#include <cstring>
+#include <future>
+#include <string>
+#include <thread>
+
+// clang-format off
+// MSVC: include <math.h> here otherwise PI gets defined twice
+#include <cmath>
+
+#ifdef _WIN32
+    #pragma comment(lib, "Ws2_32.lib")
+
+    // winsock2 must be included before windows.h
+    #include <winsock2.h>
+    #include <ws2tcpip.h>
+
+    #define LAST_SOCKET_ERROR() WSAGetLastError()
+    #undef EWOULDBLOCK
+    #define EWOULDBLOCK WSAEWOULDBLOCK
+    #ifndef SHUT_RD
+        #define SHUT_RD SD_RECEIVE
+    #endif
+    #ifndef SHUT_WR
+        #define SHUT_WR SD_SEND
+    #endif
+    #ifndef SHUT_RDWR
+        #define SHUT_RDWR SD_BOTH
+    #endif
+    #define FLAG_NO_PIPE 0
+#else
+    #include <arpa/inet.h>
+    #include <cerrno>
+    #include <fcntl.h>
+    #include <net/if.h>
+    #include <netdb.h>
+    #include <netinet/in.h>
+    #include <netinet/tcp.h>
+    #include <sys/ioctl.h>
+    #include <sys/socket.h>
+    #include <unistd.h>
+    using SOCKET = int32_t;
+    #define SOCKET_ERROR -1
+    #define INVALID_SOCKET -1
+    #define LAST_SOCKET_ERROR() errno
+    #define closesocket ::close
+    #define ioctlsocket ioctl
+    #if defined(__linux__)
+        #define FLAG_NO_PIPE MSG_NOSIGNAL
+    #else
+        #define FLAG_NO_PIPE 0
+    #endif // defined(__linux__)
+
+#endif // _WIN32
+// clang-format on
+
+#include "../Console.h"
+#include "Socket.h"
+
+namespace OpenLoco::Network
+{
+    constexpr auto CONNECT_TIMEOUT = std::chrono::milliseconds(3000);
+
+// RAII WSA initialisation needed for Windows
+#ifdef _WIN32
+    class WSA
+    {
+    private:
+        bool _isInitialised{};
+
+    public:
+        bool isInitialised() const
+        {
+            return _isInitialised;
+        }
+
+        bool initialise()
+        {
+            if (!_isInitialised)
+            {
+                Console::logVerbose("WSAStartup()");
+                WSADATA wsaData;
+                if (WSAStartup(MAKEWORD(2, 2), &wsaData) != 0)
+                {
+                    Console::error("Unable to initialise winsock.");
+                    return false;
+                }
+                _isInitialised = true;
+            }
+            return true;
+        }
+
+        ~WSA()
+        {
+            if (_isInitialised)
+            {
+                Console::logVerbose("WSACleanup()");
+                WSACleanup();
+                _isInitialised = false;
+            }
+        }
+    };
+
+    static bool InitialiseWSA()
```

Comment:
`initaliseWSA` could be an idea to have a look through whole file to make sure it conforms to coding style

Reply:
Will do, a lot of  it is copied from OpenRCT2 - so that convention sneaked in.